compiler/src/iree/compiler/Codegen/Common/ResolveSwizzleHints.cpp

```diff
@@ -164,6 +188,18 @@ static void resolveHintOp(RewriterBase &rewriter,
       stores.push_back(store);
       continue;
     }
+    if (auto gatherToLDSOp = dyn_cast<amdgpu::GatherToLDSOp>(user)) {
+      VectorType transferType =
+          dyn_cast<VectorType>(gatherToLDSOp.getTransferType());
+      if (!transferType || transferType.getRank() != 1) {
+        return;
+      }
+      if (transferType.getShape()[0] != accessWidth) {
```

Comment:
This should also compare element type bitwidth. Also it would be nice to handle cases where the transfer type isn't a vector. Really all we care about is that the swizzle bitwidth and transfer bitwidth is the same, so you can do `accessWidth * memref.getElementType().getBitwidth() == gatherToLDSOp.getTransferType().getBitwidth()` instead of comparing shapes.